src/activity_handlers.cpp

```diff
@@ -1330,9 +1330,9 @@ void activity_handlers::fill_liquid_do_turn( player_activity *act, player *p )
                 break;
         }
 
-        static const auto volume_per_turn = units::from_liter( 4 );
-        const int charges_per_turn = std::max( 1, liquid.charges_per_volume( volume_per_turn ) );
-        liquid.charges = std::min( charges_per_turn, liquid.charges );
+        static const units::volume volume_per_second = units::from_liter( 4 / 6 );
```

Comment:
You changed this to 1 liter per second. `4 / 6` is `1` in C++ (both operands are integers, so integer division is performed). If you wan't floating points, you have to explicitly make one operand a floating point value: `( 4. / 6 )`

Reply:
It's 0 actually.

Reply:
Ahhh my brain, I should have noticed that but for some reason I thought it was ok because of the units interface.